cmflib/merger.py

```diff
@@ -1,73 +1,292 @@
+                        "output",
+                        props,
+                        custom_properties=custom_props,
+                    )
                 elif artifact_type == "Model" and event_type == 3:
-                    uri = event['artifact']['uri']
                     props["uri"] = uri
-                    # model_framework = props['model_framework']
-                    # model_type = props['model_type']
-                    # model_name = props['model_name']
-                    cmf_class.log_model_with_version(path=artifact_name, event="input",props=props,
-                     custom_properties=props)
+                    cmf_class.log_model_with_version(
+                        path=artifact_name,
+                        event="input",
+                        props=props,
+                        custom_properties=props,
+                    )
                 elif artifact_type == "Model" and event_type == 4:
-                    uri = event['artifact']['uri']
                     props["uri"] = uri
-                    # model_framework = props['model_framework']
-                    # model_type = props['model_type']
-                    # model_name = props['model_name']
-                    # #print(props[""])
-                    # #props["uri"] = props
-                    # print(model_framework)
-                    # print(model_type)
-                    # print(model_name)
-                    # print(artifact_name)
-                    # print(type(artifact_name))
-                    cmf_class.log_model_with_version(path=artifact_name, event="output", props=props,
-                     custom_properties=props)
+                    cmf_class.log_model_with_version(
+                        path=artifact_name,
+                        event="output",
+                        props=props,
+                        custom_properties=props,
+                    )
                 elif artifact_type == "Metrics":
+                    # print(props,'parse')
+                    # cmf_class.log_execution_metrics_with_uuid(props, custom_props)
                     cmf_class.log_execution_metrics(artifact_name, custom_props)
                 else:
                     pass
 
 
+# mlmd, pulled from cmf-server is created from metadata passed in Json format for execution_id specified by user
+def pull_execution_to_mlmd(mlmd_data, path_to_store, pipeline_name, exec_id):
+    mlmd_data = json.loads(mlmd_data)
+    pipelines = mlmd_data["Pipeline"]
+    # print(type(pipelines))
+    pipeline = pipelines[0]
+    # print(type(pipeline))
+    pipeline_name = pipeline["name"]
+    # print(type(pipeline_name))
+    execution = {}
+    _ = None
+    cmf_class = cmf.Cmf(filename=path_to_store, pipeline_name=pipeline_name)
+    for stage in mlmd_data["Pipeline"][0]["stages"]:
+        # print(stage['name'])
+        for execution in stage["executions"]:
+            if execution["id"] == int(exec_id):
```

Comment:
After the execution id that we are interested is found, break from the loop

Reply:
Done